gnowsys-ndf/gnowsys_ndf/ndf/views/mio.py

```diff
@@ -0,0 +1,432 @@
+import smtplib
+import email
+import os
+import io
+import time
+import sys
+import imaplib
+import shutil
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.header import Header
+from email.mime.message import MIMEMessage
+import urllib
+from HTMLParser import HTMLParser
+from gnowsys_ndf.ndf.models import * 
+from django.http import HttpRequest
+from django.contrib.auth.models import User
+from gnowsys_ndf.settings import GSTUDIO_LOGS_DIR_PATH
+from gnowsys_ndf.ndf.views.filehive import *
+from gnowsys_ndf.ndf.views.methods import create_grelation
+from gnowsys_ndf.ndf.views.methods import get_group_name_id
+from django.shortcuts import render
+
+error_message = {"Page name required":1,
+				"Page Content required":2,
+				"User details required":3,
+				"already exists":4,
+				"User does not exist":5,
+				"User authenticated":6,
+				"User not a member of this group":7,
+				"Group does not exist":8}
+
+def create_page(**kwargs): 
+	p = node_collection.collection.GSystem()
+
+	if kwargs.has_key('group_name'):
+		group_name = kwargs.get('group_name','home')
+	else:
+		group_name = 'home'
+	group_name = unicode(group_name)
+
+	if kwargs.has_key('name'):
+		name = kwargs.get('name','')
+	else:
+		return error_message["Page name required"]
+	name = unicode(name)
+
+	if kwargs.has_key('content'):
+		content = kwargs.get('content','')
+	else:
+		return error_message["Page Content required"]
+	content = unicode(content)
+
+	if kwargs.has_key('created_by'):
+		created_by = kwargs.get('created_by','')
+	else:
+		return error_message["User details required"]
+
+	m_id = kwargs.get('m_id','')
+
+	gst_page = node_collection.one({'_type':u'GSystemType','name':u'Page'})
+	gst_group = node_collection.one({'_type':u'Group','name':group_name})
+
+	available_nodes = node_collection.find({'_type': u'GSystem', 'member_of': ObjectId(gst_page._id),
+		'group_set': ObjectId(gst_group._id)})
+
+	nodes_list = []
+	for each in available_nodes:
+		nodes_list.append(str((each.name).strip().lower()))
+
+	if name in nodes_list:
+		return error_message["already exists"]
+	else:
+		p.fill_gstystem_values(name = name,member_of=[gst_page._id],created_by = created_by,
+			content = content,group_set=[gst_group._id])
+		p.origin.append({'mio':m_id})
+		try:
+			p.save()
+		except:
+			print "in except"
+		return str(p._id);
+
+def send_page(to_user,page_name,page_content,subject,m_id,ref):
+        msg = MIMEMultipart('alternative')
+        msg["Message-ID"] = email.utils.make_msgid()
+        msg["In-Reply-To"] = m_id
+
+        if(ref==None):
+                msg["References"] = m_id
+        else:
+                msg["References"] = ref+"\n"+m_id
+
+        msg['Subject'] = subject
+        msg['From'] = "ps.mio.bits@gmail.com"
```

Comment:
Modify: `msg['From'] = "ps.mio.bits@gmail.com"` with following steps to make it standardize and easy to configure.

1. Add new configuration variable in `settings.py` file: `GSTUDIO_MIO_FROM_EMAIL = ''` and `GSTUDIO_MIO_FROM_EMAIL_PASSWORD = ''`.
2. Let admin to overwrite it in `local_settings.py`.
3. Import `GSTUDIO_MIO_FROM_EMAIL` and `DEFAULT_FROM_EMAIL` from setting.py.
4. Give priority/preference to `GSTUDIO_MIO_FROM_EMAIL` and keep `DEFAULT_FROM_EMAIL` in fallback.
   - If `GSTUDIO_MIO_FROM_EMAIL` is non empty, assign it to `msg['From']`, else use `DEFAULT_FROM_EMAIL`